trpcultivate_phenotypes/src/TripalCultivateValidator/ValidatorTraits/FileTypes.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\trpcultivate_phenotypes\TripalCultivateValidator\ValidatorTraits;
+
+/**
+ * Provides setters focused for setting a file types and file media type (MIME) 
+ * supported by the importer, and getter to retrieve the set value.
+ */
+trait FileTypes {
+  /**
+   * The key used by the setter method to create a file types element 
+   * in the context array, as well as the key used by the getter method 
+   * to reference and retrieve the file types element value.
+   * 
+   * @var string
+   */
+  private string $trait_key = 'file_types';
+
+  /**
+   * Sets a file types with the file media type.
+   *
+   * @param array $extensions
+   *   An array of file extensions (file types) the importer expects the file extension of the data file.
+   * 
+   * @return void
+   * 
+   * @throws \Exception
+   *  - Types array is an empty array.
+   *  - Unsupported extension (cannot resolve mime type using the type-mime mapping array).
+   */
+  public function setFileTypes(array $extensions) {
+    
+    // Extensions array must have a element.
+    if (empty($extensions)) {
+      throw new \Exception('The File Types Trait requires an array of file extensions and must not be empty.');
+    }
+    
+    // File extension and file media type (mime type).
+    $file_type_to_mime = [
+      'tsv'  => 'text/tab-separated-values',
+      'csv'  => 'text/csv',
+      'txt'  => 'text/plain',
+
+      // Add any additional valid file types here, 
+      // while ensuring file types support use of a delimiter.
+    ];
+
+    // Resolve the types to the correct mime type.
+    $file_types = [];
+    $unresolved = [];
+
+    foreach($extensions as $type) {
+      if (!isset($file_type_to_mime[ $type ])) {
+        array_push($unresolved, $type);
+        continue;
+      }
+      
+      $file_types[ $type ] = $file_type_to_mime[ $type ];
```

Comment:
Instead of saving a mapping of supported mime types, what if we save supported mime types separately from supported file extensions. This makes it easier to use these in your validator.

Specifically we would do `$mime_types = array_merge($mime_types, $file_type_to_mime[$type]);` here and then at the end of this setter we would have `$this->context['mime_types'] = $mime_types` and `$this->context['file_extensions'] = $file_types;`

Reply:
Implemented using the setSupportedMimeTypes() setter method.